src/material-experimental/mdc-form-field/form-field.ts

```diff
@@ -501,4 +551,52 @@ export class MatFormField implements AfterViewInit, OnDestroy, AfterContentCheck
       this._control.setDescribedByIds(ids);
     }
   }
+
+  /**
+   * Updates the horizontal offset of the label in the outline appearance. In the outline
+   * appearance, the notched-outline and label are not relative to the infix container because
+   * the outline intends to surround prefixes, suffixes and the infix. This means that the
+   * floating label by default overlaps prefixes in the docked state. To avoid this, we need to
+   * horizontally offset the label by the width of the prefix container. The MDC text-field does
+   * not need to do this because they use a fixed width for prefixes. Hence, they can simply
+   * incorporate the horizontal offset into their default text-field styles.
+   */
+  private _updateOutlineLabelOffset() {
+    if (!this._hasOutline() || !this._prefixContainer || !this._floatingLabel) {
+      return;
+    }
+    // If the form-field is not attached to the DOM yet (e.g. in a tab), we defer
+    // the label offset update until the zone stabilizes.
+    if (!this._isAttachedToDOM()) {
+      this._needsOutlineLabelOffsetUpdateOnStable = true;
+      return;
+    }
+
+    const floatingLabel = this._floatingLabel.element;
+    const prefixContainer = this._prefixContainer.nativeElement as HTMLElement;
+    // If the directionality is RTL, the x-axis transform needs to be inverted. This
+    // is because `transformX` does not change based on the page directionality.
+    const labelHorizontalOffset =
+      (this._dir.value === 'rtl' ? -1 : 1) * prefixContainer.getBoundingClientRect().width;
+
+    // Update the transform the floating label to account for the prefix container. Note
+    // that we do not want to overwrite the default transform for docked floating labels.
+    floatingLabel.style.transform =
+        `${FLOATING_LABEL_DEFAULT_DOCKED_TRANSFORM} translateX(${labelHorizontalOffset}px)`;
+    this._needsOutlineLabelOffsetUpdateOnStable = false;
+  }
+
+  /** Checks whether the form field is attached to the DOM. */
+  private _isAttachedToDOM(): boolean {
```

Comment:
```suggestion
  private _isAttachedToDom(): boolean {
```

Reply:
Yep. Will update (I took this from the original form-field)